weather_aplication/SchowWeatherViewController.swift

```diff
@@ -0,0 +1,118 @@
+import SnapKit
+import UIKit
+
+struct CellData {
+    let hour: String
+    let temperature: Double
+    let pressure: Double
+    let windSpeed: Double
+}
+
+class ShowWeatherViewController: UIViewController{
+    private let kSideMargin: CGFloat = 10
+    private let kTopMargin: CGFloat = 20
+    
+    private var longitude: Double = 0
+    private var latitude: Double = 0
+    
+    private let cityLabel = UILabel()
+    private let weatherTableView = UITableView()
+    
+    private var dataArray: [CellData] = []
+    
+    func getCityName() {
+        guard let queryURL =  URL(string:"https://api.open-meteo.com/v1/forecast?latitude=\(latitude)&longitude=\(longitude)&hourly=temperature_2m,surface_pressure,windspeed_10m") else { return }
+        let session = URLSession.shared
+        
+        session.dataTask(with: queryURL, completionHandler: { [weak self] data, response, error -> Void in
+            
+            if (error != nil) {
+                let alert = UIAlertController(title: "Error", message: error?.localizedDescription, preferredStyle: .alert)
+                alert.addAction(UIAlertAction(title: "OK", style: .default))
+                self?.present(alert, animated: true)
+            }
+            
+            if let httpResponse = response as? HTTPURLResponse {
+                if httpResponse.statusCode == 200 {
+                    do {
+                        guard let data = data else { return }
+                        let jsonResult = try JSONDecoder().decode(OpenMeteoJSON.self, from: data)
+                        self?.decodeJson(jsonResult: jsonResult)
+                        DispatchQueue.main.async {
+                            self?.weatherTableView.reloadData()
+                        }
+                    } catch let error {
+                        print(error)
+                    }
+                }
+            }
+            
+        }).resume()
+    }
+    
+    func decodeJson(jsonResult: OpenMeteoJSON) {
+        let datas = jsonResult.hourly.time.count
```

Comment:
Nazwa zmiennej nie odpowiada temu co to jest, bardziej pasuje elementsCount lub coś w tym stylu